library/Icinga/Web/RememberMe.php

```diff
@@ -92,19 +99,18 @@ public static function fromCookie()
             ->setKey(hex2bin($rs->passphrase))
             ->setIV($iv);
 
-        if (version_compare(PHP_VERSION, AesCrypt::GCM_SUPPORT_VERSION, '>=')) {
-            $tag = array_pop($data);
-
-            if (empty($data)) {
-                $rememberMe->aesCrypt = (new AesCrypt())
-                    ->setMethod('AES-128-CBC')
-                    ->setKey(hex2bin($rs->passphrase))
-                    ->setIV($iv);
-
-                $data[0] = $tag; // encryptedPass
-            } else {
-                $rememberMe->aesCrypt->setTag(base64_decode($tag));
+        if (count($data) > 1) {
+            // php downgrade
+            if (! $rememberMe->aesCrypt->isAESupported()) {
+                throw new RuntimeException(sprintf(
+                    "The given decryption method is not supported in php version '%s'",
+                    PHP_VERSION
+                ));
```

Comment:
I'd rather throw this in `setTag()`. Setting a tag for `AesCrypt` on PHP < 7.1 should not be possible.